### PR TITLE
refactor: go through clj-kondo warnings

### PR DESCRIPTION
**Pull request by yqrashawn**
### Summary

go through clj-kondo errors and warnings

### Review notes
check comments in code

status: ready

### CODE REVIEW

File: src/quo/react.cljs [hunk 1]
```diff
@@ -25,21 +25,21 @@
   (-hash [o] (goog/getUid o))
 
   cljs.core/IDeref
-  (-deref [o]
+  (-deref [_o]
```
Comment (yqrashawn) [inline src/quo/react.cljs:28]: unused-binding

File: src/status_im/core.cljs [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
             [status-im.utils.snoopy :as snoopy]
-            [status-im.utils.config :as config]
-            [status-im.utils.universal-links.core :as utils.universal-links]
-            [status-im.i18n.i18n :as i18n]
-            [status-im.ui.components.react :as react]
-            [status-im.async-storage.core :as async-storage]
-            ["react-native" :refer (DevSettings LogBox)]
-            ["react-native-languages" :default react-native-languages]
-            ["react-native-shake" :as react-native-shake]))
+            [status-im.utils.universal-links.core :as utils.universal-links]))
```
Comment (yqrashawn) [inline src/status_im/core.cljs:22]: ns reorder

File: src/status_im/ethereum/eip681_test.cljs [hunk 3]
```diff
@@ -139,10 +139,10 @@
     (is (.equals (money/ether->wei (money/bignumber 1)) value))
     (is (= :ETH symbol))
     (is (= "0x89205a3a3b2a69de6dbf7f01ed13b2108b2c43e7" address)))
-  (is (= (eip681/extract-request-details {:address "0x744d70fdbe2ba4cf95131626614a1763df805b9e" :chain-id 1 :function-name "unknown"}
-                                         {:address "0x744d70fdbe2ba4cf95131626614a1763df805b9e" :chain-id 1 :function-name "unknown"})))
-  (let [{:keys [value symbol address]} (eip681/extract-request-details {:address "0x744d70fdbe2ba4cf95131626614a1763df805b9e" :chain-id 1
-                                                                        :function-name "transfer" :function-arguments {:uint256 1000 :address "0x89205a3a3b2a69de6dbf7f01ed13b2108b2c43e7"}}
+  (is (= (eip681/extract-request-details {:address "0x744d70fdbe2ba4cf95131626614a1763df805b9e" :chain-id 1 :function-name "unknown"} {})
```
Comment (yqrashawn) [inline src/status_im/ethereum/eip681_test.cljs:142]: related commit https://github.com/status-im/status-mobile/commit/45dae49fc0964aa49d8221b6de07409895bc9f25#diff-8e3e99be89a988de9f8238b3c65568e04bd433e2eae50436070d66d2c0f611c6R138

File: src/status_im/utils/priority_map.cljs [hunk 4]
```diff
@@ -9,15 +9,16 @@
 (deftype PersistentPriorityMap [priority->set-of-items item->priority meta keyfn ^:mutable __hash]
   IPrintWithWriter
   (-pr-writer [coll writer opts]
-    (let [pr-pair (fn [keyval] (pr-sequential-writer writer pr-writer "" " " "" opts keyval))]
+    (let [pr-pair (fn [keyval]
+                    (pr-sequential-writer writer #'cljs.core/pr-writer "" " " "" opts keyval))]
```
Comment (yqrashawn) [inline src/status_im/utils/priority_map.cljs:13]: related issue https://github.com/clj-kondo/clj-kondo/issues/1558#issuecomment-1020036716

File: shadow-cljs.edn [hunk 5]
```diff
@@ -91,7 +91,8 @@
                               :source-map    false
                               ;; needed because we use deref in tests
                               :static-fns    false
-                              :optimizations :simple}}
+                              :optimizations :simple
+                              :infer-externs true}}
```
Comment (yqrashawn) [inline shadow-cljs.edn:95]: Get rid of cannot infer warning

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**